### PR TITLE
add argoCD application yaml

### PR DESCRIPTION
**Pull request by matthyx**
- revert to the Helm hook commit
- implement `syncPolicy` as described in http://www.zakariaamine.com/2023-06-18/argocd-with-helm-hooks/
- use `kubectl patch` to keep existing values in cloud config (especially `clusterData`)

### CODE REVIEW

File: ArgoCDApplication.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    helm:
+      valueFiles:
+        - values.yaml
+      parameters:
+        - name: account
+          value: '' # add account ID
+        - name: clusterName
+          value: '' # add cluster name
+        - name: server
+          value: 'api.armosec.io'
```
Comment (dwertent) [inline ArgoCDApplication.yaml:23]: We should not include `api.armosec.io` by default

**Review by dwertent — CHANGES_REQUESTED** [review]

File: ArgoCDApplication.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  name: kubescape
+spec:
+  destination:
+    name: ''
+    namespace: kubescape
+    server: 'https://kubernetes.default.svc' # change to your server
+  source:
+    path: charts/kubescape-operator
+    repoURL: 'https://github.com/kubescape/helm-charts'
+    targetRevision: argo
```
Comment (matthyx) [inline ArgoCDApplication.yaml:13]:
```suggestion
    targetRevision: HEAD
```

Comment (dwertent) [reply ArgoCDApplication.yaml:13]: What is blocking you from changing this?

Comment (matthyx) [reply ArgoCDApplication.yaml:13]: I wanted to test it with Akinox this afternoon

**Pull request merged**